### PR TITLE
feat(AWS IAM): Add support for `iam.role.name` definition

### LINKED ISSUE #8887: Cannot specify name of default AWS IAM Role
### Use case description

Currently the default AWS IAM Role is named with a static naming convention that cannot be customized,

https://github.com/serverless/serverless/blob/61dd3bde8d17cdd995fdd27259a689d12bee1e42/lib/plugins/aws/lib/naming.js#L93-L105

It would be beneficial to many projects to provide a name (e.g. the service name and/or stage name combine to a long string, opening potential issues with the AWS Role name length limit of 64 characters. 

Allowing the name to be customized would solve the issue as reported in #4918, #2856.

Similar solutions appear to have been proposed and either disagreed on or otherwise abandoned (#7821).

### Proposed solution

Introduce a **new** configuration item `provider.iamRoleName` to allow for specifying a name other than the default generated one.

The new item would should be backwards-compatibile as it will be optional and continue to default to the existing name scheme.

### PR DESCRIPTION
**Pull request by Enase**
Inspired by https://github.com/serverless/serverless/pull/8888

Fixes #8887
Closes: #8887

### CODE REVIEW

File: lib/plugins/aws/provider.js [hunk 1]
```diff
@@ -828,6 +828,7 @@ class AwsProvider {
                     {
                       type: 'object',
                       properties: {
+                        name: { type: 'string' },
```
Comment (pgrzesik) [inline lib/plugins/aws/provider.js:831]: Let's ensure a bit better validation here following the valid values for RoleName: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-iam-role.html#cfn-iam-role-rolename

File: test/unit/lib/plugins/aws/package/lib/mergeIamTemplates.test.js [hunk 2]
```diff
@@ -282,6 +282,29 @@ describe('lib/plugins/aws/package/lib/mergeIamTemplates.test.js', () => {
...
+                },
+              },
+            },
+          },
+        });
+
+        const { cfTemplate, awsNaming } = result;
+        const iamRoleLambdaResource = cfTemplate.Resources[awsNaming.getRoleLogicalId()];
+        const roleName = awsNaming.getRoleName();
+        expect(roleName).to.be.eq(customRoleName);
```
Comment (pgrzesik) [inline test/unit/lib/plugins/aws/package/lib/mergeIamTemplates.test.js:304]: let's skip this check as it's already tested as a part of `naming.test.js` tests

**Review by pgrzesik — CHANGES_REQUESTED** [review]
Thank you @Enase for picking this one up - great work overall :raised_hands: I just have a few small comments, please let me know what do you think

File: test/unit/lib/plugins/aws/package/lib/mergeIamTemplates.test.js (hunk 2 above)
Comment (Enase) [reply test/unit/lib/plugins/aws/package/lib/mergeIamTemplates.test.js]: :white_check_mark:

File: lib/plugins/aws/provider.js (hunk 1 above)
Comment (Enase) [reply lib/plugins/aws/provider.js]: :white_check_mark:

File: test/unit/lib/plugins/aws/package/lib/mergeIamTemplates.test.js [hunk 3]
```diff
@@ -282,6 +282,29 @@ describe('lib/plugins/aws/package/lib/mergeIamTemplates.test.js', () => {
...
+              iam: {
+                role: {
+                  name: customRoleName,
+                },
+              },
+            },
+          },
+        });
+
+        const { cfTemplate, awsNaming } = result;
```
[the remark this answers is not in the document]
Comment (Enase) [reply test/unit/lib/plugins/aws/package/lib/mergeIamTemplates.test.js]: :white_check_mark:

**Comment by Enase** [comment]
@pgrzesik pls review..

**Review by pgrzesik — APPROVED** [review]
Thank you @Enase - great work :+1:

**Pull request merged**